### PR TITLE
Fix properties of unset special tokens in non verbose mode

### LINKED ISSUE #17796: Properties of unset special tokens return the string 'None' in non verbose Tokenizers
### System Info

```shell
- `transformers` version: 4.20.0
- Platform: Linux-5.4.0-120-generic-x86_64-with-glibc2.27
- Python version: 3.8.0
- Huggingface_hub version: 0.7.0
- PyTorch version (GPU?): 1.11.0+cu102 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no
```

### Who can help?

@SaulLu

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

For example, `MarianTokenizer` does not set `bos_token`. The corresponding property returns the string `'None'`:

```python
>>> import transformers
>>> tokenizer = transformers.MarianTokenizer.from_pretrained("Helsinki-NLP/opus-mt-en-de", verbose=False)
>>> tokenizer.bos_token
'None'
```

### Expected behavior

```shell
The property should return None, not the string 'None'.
```

### PR DESCRIPTION
**Pull request by guillaumekln**
# What does this PR do?

Fixes #17796.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

@n1t0, @LysandreJik

### CODE REVIEW

File: tests/test_tokenization_common.py [hunk 1]
```diff
@@ -578,6 +579,25 @@ def test_tokenizers_common_ids_setters(self):
                 self.assertListEqual(getattr(tokenizer, "additional_special_tokens"), [token_to_test_setters])
                 self.assertListEqual(getattr(tokenizer, "additional_special_tokens_ids"), [token_id_to_test_setters])
 
+    @parameterized.expand([(True,), (False,)])
+    def test_tokenizers_common_properties_unset(self, verbose):
```
Comment (SaulLu) [inline tests/test_tokenization_common.py:583]: A small suggestion to rename the test as I think it is specifically about "special tokens". WDYT? :blush: 

```suggestion
    def test_tokenizers_special_tokens_properties_unset(self, verbose):
```

**Review by SaulLu — APPROVED** [review]
Thank you very much for proposing a fix to the problem you have identified! 

Your solution seems perfect, I just left a micro comment about the test title.

@LysandreJik , @sgugger , @patrickvonplaten or @patil-suraj  would you also like to review this PR too?

File: tests/test_tokenization_common.py (hunk 1 above)
Comment (guillaumekln) [reply tests/test_tokenization_common.py]: Sure, let's rename the test case.

**Review by patrickvonplaten — APPROVED** [review]
Thanks for the PR @guillaumekln

**Pull request merged**